core/include/gnuradio-4.0/CircularBuffer.hpp

```diff
@@ -715,32 +764,23 @@ class CircularBuffer
             return _isRangeConsumed;
         }
 
-        template<SpanReleasePolicy policy = SpanReleasePolicy::Terminate>
-        [[nodiscard]] constexpr auto get(const std::size_t nRequested) const noexcept -> ConsumableInputRange<U, policy> {
-            _isRangeConsumed = false;
-            return ConsumableInputRange<U, policy>(this, buffer_index(), nRequested);
+        [[nodiscard]] constexpr bool isConsumeRequested() const noexcept {
+            return _nSamplesToConsume != std::numeric_limits<std::size_t>::max();
         }
 
-        template <bool strict_check = true>
-        [[nodiscard]] constexpr bool consume(const std::size_t nSamples = 1) noexcept {
-            if (isConsumed()) {
-                fmt::println("An error occurred: The method CircularBuffer::buffer_reader::consume() was invoked for the second time in succession, a corresponding ConsumableInputRange was already consumed.");
-                std::abort();
+        template<SpanReleasePolicy policy = SpanReleasePolicy::ProcessNone>
+        [[nodiscard]] constexpr auto get(const std::size_t nRequested = std::numeric_limits<std::size_t>::max()) const noexcept -> ConsumableInputRange<U, policy> {
+            if (isConsumeRequested()) {
+                assert(false && "The method CircularBuffer::buffer_reader::get() was invoked after consume() methods was explicitly invoked.");
             }
-
-            if constexpr (strict_check) {
-                if (nSamples <= 0) {
-                    _isRangeConsumed = true;
-                    return true;
-                }
-                if (nSamples > available()) {
-                    _isRangeConsumed = true;
-                    return false;
-                }
+            std::size_t nSamples = std::min(available(), nRequested);
```

Comment:
I'd suggest to make this call to `available()` conditional only if `nRequested != std::numeric_limits<std::size_t>::max()`. The call to available involves atomics which are not cache friendly.

Reply:
I think it is unavoidable: `nRequested == std::numeric_limits<std::size_t>::max()` means return all available samples -> `nSamples = available()`.

Reply:
the default parameter is `nRequested = std::numeric_limits<std::size_t>::max()` which basically returns number of available samples.

Reply:
One can introduce stick_check template parameter, and if it is false one can remove check with available()